recipes/llvm-core/all/patches/11x/0000-cmake-dependencies.patch

```diff
@@ -0,0 +1,118 @@
+diff --git a/cmake/config-ix.cmake b/cmake/config-ix.cmake
+index 90e5d327c75..c8bcb1afb76 100644
+--- a/cmake/config-ix.cmake
++++ b/cmake/config-ix.cmake
+@@ -5,6 +5,7 @@ endif()
+ 
+ include(CheckIncludeFile)
+ include(CheckLibraryExists)
++include(CMakePushCheckState)
+ include(CheckSymbolExists)
+ include(CheckFunctionExists)
+ include(CheckStructHasMember)
+@@ -56,13 +57,27 @@ check_include_file(sys/types.h HAVE_SYS_TYPES_H)
+ check_include_file(termios.h HAVE_TERMIOS_H)
+ check_include_file(unistd.h HAVE_UNISTD_H)
+ check_include_file(valgrind/valgrind.h HAVE_VALGRIND_VALGRIND_H)
+-check_include_file(zlib.h HAVE_ZLIB_H)
++
++if(LLVM_ENABLE_ZLIB)
++    find_package(ZLIB REQUIRED)
++    cmake_push_check_state()
++    list(APPEND CMAKE_REQUIRED_INCLUDES ${ZLIB_INCLUDE_DIRS})
++    check_include_file(zlib.h HAVE_ZLIB_H)
++    cmake_pop_check_state()
++endif()
++
+ check_include_file(fenv.h HAVE_FENV_H)
+ check_symbol_exists(FE_ALL_EXCEPT "fenv.h" HAVE_DECL_FE_ALL_EXCEPT)
+ check_symbol_exists(FE_INEXACT "fenv.h" HAVE_DECL_FE_INEXACT)
+ 
+ check_include_file(mach/mach.h HAVE_MACH_MACH_H)
+-check_include_file(histedit.h HAVE_HISTEDIT_H)
++if(LLVM_ENABLE_LIBEDIT)
++    find_package(editline REQUIRED)
++    cmake_push_check_state()
++    list(APPEND CMAKE_REQUIRED_INCLUDES ${editline_INCLUDE_DIRS})
++    check_include_file(histedit.h HAVE_HISTEDIT_H)
++    cmake_pop_check_state()
++endif()
+ check_include_file(CrashReporterClient.h HAVE_CRASHREPORTERCLIENT_H)
+ if(APPLE)
+   include(CheckCSourceCompiles)
+@@ -118,24 +133,21 @@ endif()
+ # Don't look for these libraries if we're using MSan, since uninstrumented third
+ # party code may call MSan interceptors like strlen, leading to false positives.
+ if(NOT LLVM_USE_SANITIZER MATCHES "Memory.*")
+-  set(HAVE_LIBZ 0)
+   if(LLVM_ENABLE_ZLIB)
+-    foreach(library z zlib_static zlib)
+-      string(TOUPPER ${library} library_suffix)
+-      check_library_exists(${library} compress2 "" HAVE_LIBZ_${library_suffix})
+-      if(HAVE_LIBZ_${library_suffix})
+-        set(HAVE_LIBZ 1)
+-        set(ZLIB_LIBRARIES "${library}")
+-        break()
+-      endif()
+-    endforeach()
++    cmake_push_check_state()
++    list(APPEND CMAKE_REQUIRED_INCLUDES ${ZLIB_INCLUDE_DIRS})
++    list(APPEND CMAKE_REQUIRED_LIBRARIES ${ZLIB_LIBRARIES})
++    check_symbol_exists(compress2 zlib.h HAVE_LIBZ)
++    cmake_pop_check_state()
+   endif()
+ 
+-  # Don't look for these libraries on Windows.
+-  if (NOT PURE_WINDOWS)
```

Comment:
Because conan makes it easy to do so. I assume that the reason LLVM does not check for these libraries on Windows is dues to the difficulty in providing them. As conan allows us to provide them, I see no reason not to do so.